src/kernel/include/shared/cos_types.h

```diff
@@ -840,8 +848,8 @@ static inline void
 cos_mem_fence(void)
 { __asm__ __volatile__("mfence" ::: "memory"); }
 
-// ncpu * 16 (or max 256) entries. can be increased if necessary.
-#define COS_THD_INIT_REGION_SIZE (((NUM_CPU*16) > (1<<8)) ? (1<<8) : (NUM_CPU*16))
+/* 256 entries. can be increased if necessary */
+#define COS_THD_INIT_REGION_SIZE (1<<8)
```

Comment:
You changed it to 256, not 32/64, right?

It is OK to make it higher.  But I wonder if we want it to increase with increasing number of cores.  Perhaps not, and we actually just have a backup linked list if the array is not large enough.

OK, you can just leave it.